### PR TITLE
[SPARK-47857][SQL] Utilize `java.sql.RowId.getBytes` API directly for UTF8String

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Utilize java.sql.RowId.getBytes API directly to simplify `RowId.toString.getBytes` to RowId.getBytes.

### Why are the changes needed?

To simplify code and might get some performance gain by visiting underly bytes array directly instead of copying back and forth.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

```
SPARK-32992: map Oracle's ROWID type to StringType (721 milliseconds)
SPARK-44885: query row with ROWID type containing NULL value (821 milliseconds)
```

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala [hunk 1]
```diff
@@ -467,12 +467,8 @@ object JdbcUtils extends Logging with SQLConfHelper {
 
     case StringType if metadata.contains("rowid") =>
       (rs: ResultSet, row: InternalRow, pos: Int) =>
-        val rawRowId = rs.getRowId(pos + 1)
-        if (rawRowId == null) {
-          row.update(pos, null)
-        } else {
-          row.update(pos, UTF8String.fromString(rawRowId.toString))
-        }
+        val id = nullSafeConvert[RowId](rs.getRowId(pos + 1), r => UTF8String.fromBytes(r.getBytes))
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala:470]: Does this assume that `rowid` column is always UTF-8?

Comment (dongjoon-hyun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala:470]: Maybe, could you add a reference Java doc link to ensure that assumption?

Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala:470]: The assumption might be practical as JDBC clients use the platform's default charset. However, I guess we cannot make such an assumption. I might close this first to see if we can apply the charset suitably for the client-server encode conversion.

Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala:470]: When I say 'practical', it means RowIds are alphanumerics

Comment (dongjoon-hyun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala:470]: Got it. Thank you for the further investigation to make this sure.

**Pull request closed without merging**